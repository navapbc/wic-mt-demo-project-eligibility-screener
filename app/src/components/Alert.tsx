import { ReactElement } from 'react'

type Props = {
  text: string
  type: 'error' | 'info' | 'success' | 'warning'
}

const Alert = (props: Props): ReactElement => {
  const { text, type } = props

  return (
    <div
      className={`usa-alert usa-alert--${type} usa-alert--no-icon`}
      role="alert"
    >
      <div className="usa-alert__body">
        <p className="usa-alert__text">{text}</p>
      </div>
    </div>
  )
}

export default Alert
