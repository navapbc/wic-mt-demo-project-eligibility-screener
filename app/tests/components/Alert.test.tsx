import { render, screen } from '@testing-library/react'

import Alert, { AlertProps, AlertTypes } from '@components/Alert'

import { testSnapshot } from '../helpers/sharedTests'

const testProps: AlertProps = {
  alertBody: 'body',
  type: 'info',
}

const alertTypes: Array<AlertTypes[]> = [
  ['error'],
  ['info'],
  ['success'],
  ['warning'],
]

it.each(alertTypes)('should match snapshot for %s alerts', (type) => {
  const element = <Alert {...testProps} type={type} />
  testSnapshot(element)
})

it.each(alertTypes)(
  'should have the correct css class for %s alerts',
  (type) => {
    const element = <Alert {...testProps} type={type} />
    render(element)

    const alert = screen.getByRole('alert')
    expect(alert).toHaveClass(`usa-alert--${type}`)
  }
)

it('should display with no icon if the icon prop is undefined', () => {
  const element = <Alert {...testProps} />
  render(element)

  const alert = screen.getByRole('alert')
  expect(alert).toHaveClass(`usa-alert--no-icon`)
})

it('should display with no icon if the icon prop is false', () => {
  const element = <Alert {...testProps} icon={false} />
  render(element)

  const alert = screen.getByRole('alert')
  expect(alert).toHaveClass(`usa-alert--no-icon`)
})

it('should display an icon if the icon prop is true', () => {
  const element = <Alert {...testProps} icon={true} />
  render(element)

  const alert = screen.getByRole('alert')
  expect(alert).not.toHaveClass(`usa-alert--no-icon`)
})
