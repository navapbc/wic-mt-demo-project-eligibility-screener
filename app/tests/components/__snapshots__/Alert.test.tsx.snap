// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot for error alerts 1`] = `
<div>
  <div
    class="usa-alert usa-alert--error usa-alert--no-icon"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        body
      </p>
    </div>
  </div>
</div>
`;

exports[`should match snapshot for info alerts 1`] = `
<div>
  <div
    class="usa-alert usa-alert--info usa-alert--no-icon"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        body
      </p>
    </div>
  </div>
</div>
`;

exports[`should match snapshot for success alerts 1`] = `
<div>
  <div
    class="usa-alert usa-alert--success usa-alert--no-icon"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        body
      </p>
    </div>
  </div>
</div>
`;

exports[`should match snapshot for warning alerts 1`] = `
<div>
  <div
    class="usa-alert usa-alert--warning usa-alert--no-icon"
    role="alert"
  >
    <div
      class="usa-alert__body"
    >
      <p
        class="usa-alert__text"
      >
        body
      </p>
    </div>
  </div>
</div>
`;
